### PR TITLE
use requests with attested connection pooling

### PR DESCRIPTION
**Pull request by HernanGatta**
# Overview

This PR updates the PromptGuard client to use [the changes proposed to the underlying aTLS package](https://github.com/opaque-systems/atls-python/pull/10). Namely, this PR modifies the PromptGuard client to use the `requests` library and to utilize the aTLS adapter in the aforementioned PR to in turn gain support for connection pooling, thereby greatly reducing the per-request latency.

### CODE REVIEW

**Review by ggroode — APPROVED** [review]

File: python-package/src/promptguard/promptguard_service.py [hunk 1]
```diff
@@ -2,16 +2,22 @@
...
-from atls.validators import AZ_AAS_GLOBAL_JKUS, AzAasAciValidator, Validator
+import requests
+from atls.utils.requests import HTTPAAdapter
+from atls.validators import Validator
+from atls.validators.azure.aas import PUBLIC_JKUS, AciValidator
 from promptguard.authentication import get_api_key
 from promptguard.configuration import get_server_config
 
+_session: Optional[requests.Session] = None
+_sessionLock: threading.Lock = threading.Lock()
```
Comment (octaviansima) [inline python-package/src/promptguard/promptguard_service.py]: Nit: `_session_lock` to match Python naming convention

Some comments about what these variables are used for would also help

**Review by octaviansima — APPROVED** [review]

File: python-package/src/promptguard/promptguard_service.py [hunk 2]
```diff
@@ -103,37 +109,34 @@ def _send_request_to_promptguard_service(
...
+        if _session is None:
+            _session = requests.Session()
+            _session.mount("httpa://", HTTPAAdapter(_get_default_validators()))
+
     api_key = get_api_key()
     hostname, port = get_server_config()
 
-    ctx = AttestedTLSContext(_get_default_validators())
-    conn = AttestedHTTPSConnection(hostname, ctx, port)
+    response = _session.request(
```
Comment (zizhong) [inline python-package/src/promptguard/promptguard_service.py]: This code doesn't handle the case where connection was closed (maybe by remote). Possibly need something like https://github.com/singer-io/tap-skubana/pull/6/files or better.

File: python-package/src/promptguard/promptguard_service.py (hunk 1 above)
Comment (HernanGatta) [reply python-package/src/promptguard/promptguard_service.py]: Added.

File: python-package/src/promptguard/promptguard_service.py (hunk 2 above)
Comment (HernanGatta) [reply python-package/src/promptguard/promptguard_service.py]: Added.

**Review by zizhong — APPROVED** [review]
LG

**Review by octaviansima — APPROVED** [review]

**Pull request merged**